src/conv-core/cpuaffinity.C

```diff
@@ -241,6 +241,47 @@ int CmiSetCPUAffinity(int mycore)
   return result;
 }
 
+int CmiSetCPUAffinityLogical(int mycore)
+{
+  int core = mycore;
+  if (core < 0) {
+    core = CmiHwlocTopologyLocal.num_pus + core;
+  }
+  if (core < 0) {
+    CmiError("Error: Invalid parameter to CmiSetCPUAffinityLogical: %d\n", mycore);
+    CmiAbort("CmiSetCPUAffinityLogical failed!");
```

Comment:
Could remove this since it's covered in the error message above

Reply:
Most of this function is copied from `CmiSetCPUAffinity`, including these lines. There are plenty of locations throughout the runtime that could now have multiple print calls coalesced into a single CmiAbort. For consistency, we could leave this as it is for now, and revisit it as part of broader cleanup.

Reply:
> The hwloc documentation states, OS indexes "may be completely arbitrary, non-unique, non-contiguous, not representative of logical proximity, and may depend on the BIOS configuration." If only one core was being allocated, it may not have had OS index 0, therefore any attempted use of that index would be invalid.

Right, I understand that for `interact` which was only allocating 1 core. But the autobuild job that failed with `+setcpuaffinity` had the following configuration:
```
 30 #!/bin/sh$
 31 #SBATCH -p RM-small$
 32 #SBATCH -N 2$
 33 #SBATCH --ntasks-per-node 4$
 34 #SBATCH -t 01:20:00$
 35 #SBATCH -o $output$
 36 #SBATCH -e $error$
```